### ISSUE TITLE
Alerting: failed to evaluate queries and expressions: input data must be a wide series but got type long (input refid)

### ISSUE THREAD

**Issue by redbaron** [grafana/grafana#64552]
**What happened**:

Creating alerts with InfluxDB (Flux query) datasource fails with obscure error.

**How to reproduce it (as minimally and precisely as possible)**:

Following self containing expression triggers the problem:

```
import "csv"

csvData = "
#group,false,false,false,true,true,true,true,false,false
#datatype,string,long,dateTime:RFC3339,string,string,string,string,long,string
#default,_result,,,,,,,,
,result,table,_time,container_name,mc_zone,namespace,pod_name,_value,_field
,,1,2023-03-09T17:16:02Z,app,zone1,p-project1,app-1,10037694,cpu
,,2,2023-03-09T17:16:02Z,app,zone1,p-project1,app-0,92576117,cpu
"

csv.from(csv: csvData)
```

**Anything else we need to know?**:

There were mentions of this error in issues, but they seem to be related to missing data condition, but in my case data clearly presents.

Not every influxdb query fails to create alert, some work fine. 

**Environment**:
- Grafana version: 9.4.3 on Grafana cloud

**Comment by tonypowa** [comment]
Thanks for creating this issue, @redbaron 

are there any other error logs besides this one? 

>Alerting: failed to evaluate queries and expressions: input data must be a wide series but got type long (input refid)

Try increasing [the verbosity of the Grafana server logs](https://grafana.com/docs/grafana/latest/administration/configuration/#level) to debug and note any errors. For printing to console, [set the console logs to debug](https://grafana.com/docs/grafana/latest/administration/configuration/#level-1) as well.

Thank you 👍

**Comment by redbaron** [comment]
@tonypowa , we see it on Grafana Cloud where we have no access to logs I believe.

**Comment by redbaron** [comment]
For future poor souls who have been plagued by this obsure error, here is solution to your problems:

```
# add following at the end of your alert query
|> group(columns: ["_time","_value"], mode: "except")
```

TLDR; you have to group by every column except `_time` and `_value` for a query to be accepted by Alerts.
